presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -190,6 +191,9 @@ public SubPlan createSubPlans(Session session, Plan plan, boolean forceSingleNod
         if (forceSingleNode || isForceSingleNodeOutput(session)) {
             properties = properties.setSingleNodeDistribution();
         }
+        if (isForceCoordinatorExecution(session) && canForceCoordinatorExecution(plan.getRoot())) {
```

Comment:
Doing this on the `PlanFragmeter` level seem to be a little too late. Ideally for "single node" queries no extra exchanges should be added. For example that's how we force single node execution for local testing on the planner level: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java#L543

Reply:
the target is that the  query values node can redirect return on coordinator,that can decrease once node selecting and once data exchange.
if i add planOptimizers ,I think is to delete every exchange node ,but i think it not nessary. what do you think？